pandas/tests/indexes/multi/test_setops.py

```diff
@@ -379,8 +379,7 @@ def test_union_sort_other_incomparable():
     idx = MultiIndex.from_product([[1, pd.Timestamp("2000")], ["a", "b"]])
 
     # default, sort=None
-    with tm.assert_produces_warning(RuntimeWarning):
```

Comment:
Does this sort the index now or is the behavior as before nur does not slow the warning anymore?

Reply:
Same behavior as before just without the warning.

Reply:
So we don’t sort but we also don’t give a warning? This isn’t ideal imo

Reply:
I just made another commit that restores the expectation for a warning. I found an older issue that speaks to why the error was suppressed #18363. 

`df.sort_index` currently does not warn/error when index is a MultiIndex and unsortable. I think that could be a separate PR, but I believe it would be as simple as passing True to the new `raise_if_unsortable` parameter I just added.